iOSDeviceManager/Devices/PhysicalDevice.m

```diff
@@ -717,31 +718,41 @@ - (BOOL)stageXctestConfigurationToTmpForBundleIdentifier:(NSString *)bundleIdent
     FBiOSDeviceOperator *operator = [self fbDeviceOperator];
     [operator fetchApplications];
 
-    NSString *runnerPath;
-    runnerPath = [operator applicationPathForApplicationWithBundleID:bundleIdentifier
+    NSString *runnerPath = [operator applicationPathForApplicationWithBundleID:runnerBundleIdentifier
                                                                error:error];
-    if (!runnerPath) { return NO; }
+    NSString *AUTPath = [operator applicationPathForApplicationWithBundleID:AUTBundleIdentifier error:error];
+
+    NSString *uuid = [[NSUUID UUID] UUIDString];
 
     NSString *xctestBundlePath = [self xctestBundlePathForTestRunnerAtPath:runnerPath];
-    NSString *xctestconfig = [XCTestConfigurationPlist plistWithTestBundlePath:xctestBundlePath];
+    NSString *xctestconfig = [XCTestConfigurationPlist plistWithXCTestInstallPath:xctestBundlePath
+                                                                 AUTInstalledPath:AUTPath
+                                                              AUTBundleIdentifier:AUTBundleIdentifier
+                                                              runnerInstalledPath:runnerPath
+                                                            runnerBundleIdentifier:runnerBundleIdentifier
+                                                                sessionIdentifier:uuid];
 
     NSString *tmpDirectory = [[xcappdata stringByAppendingPathComponent:@"AppData"]
                               stringByAppendingPathComponent:@"tmp"];
 
-    NSString *filename = @"DeviceAgent.xctestconfiguration";
+    NSString *filename = [uuid stringByAppendingString:@".xctestconfiguration"];
     NSString *xctestconfigPath = [tmpDirectory stringByAppendingPathComponent:filename];
-
-    if (![xctestconfig writeToFile:xctestconfigPath
-                        atomically:YES
-                          encoding:NSUTF8StringEncoding
-                             error:error]) {
+    
+    NSData *plistData = [xctestconfig dataUsingEncoding:NSUTF8StringEncoding];
+    id plist = [NSPropertyListSerialization dataWithPropertyList:plistData
+                                                          format:NSPropertyListBinaryFormat_v1_0
+                                                         options:0
+                                                           error:error];
+
+    if (![plist writeToFile:xctestconfigPath
```

Comment:
This is not work:

```
$ ideviceteststarter ...
UITests-Runner[27734:2582437] Exception decoding test configuration data:
 The data couldn’t be read because it isn’t in the correct format.
```

Just writing the `plistData` did not work either.

Can you try to use `ShellRunner` + `plutil -convert binary1`?